packages/@atjson/document/src/query.ts

```diff
@@ -1,8 +1,40 @@
+import Document from '.';
 import Annotation from './annotation';
-import Document from './index';
 
-export interface Filter {
-  [key: string]: any;
+export type DeprecatedTransform = (annotation: JoinableAnnotation) => Annotation | null;
```

Comment:
When I specify this branch in all `copilot-atjson` submodules' dependencies and run `npm run reset`, I see errors like
```
src/index.ts(221,38): error TS2345: Argument of type '(oldAnnotation: Annotation) => { type: string; start: number; end: number; attributes: { [key: string]: any; }; }' is not assignable to parameter of type 'Renaming | Transform | DeprecatedTransform'.
  Type '(oldAnnotation: Annotation) => { type: string; start: number; end: number; attributes: { [key: string]: any; }; }' is not assignable to type 'DeprecatedTransform'.
    Types of parameters 'oldAnnotation' and 'annotation' are incompatible.
      Type 'JoinableAnnotation' is not assignable to type 'Annotation'.
        Type 'AnnotationJoin' is not assignable to type 'Annotation'.
          Property 'type' is missing in type 'AnnotationJoin'.
```
because existing usages of `Query#map`.

I suspect some nuance of TS and maybe we just need to throw another allowed type in somewhere (`| <something>`)

rep steps:
 1. check out `latest` of `copilot-atjson`
 1. replace all submodules' package.json version of `atjson` with this branch
 1. in repo root run `npm i`
 1. in repo root run `npm run reset`